common/model-views.cpp

```diff
@@ -4166,11 +4170,58 @@ namespace rs2
         }
         return clicked;
     }
-    bool device_model::prompt_toggle_advanced_mode(bool enable_advanced_mode, const std::string& message_text, std::vector<std::string>& restarting_device_info, viewer_model& view, ux_window& window)
+
+    // Create a process windows with process details from the caller,
+    // and close button activated by the caller
+    bool process_model(const std::string& title, const std::string& process_topic_text, const std::string& process_status_text , bool enable_close, ux_window& window)
+    {
+        ImGui_ScopePushFont(window.get_large_font());
+        ImGui_ScopePushStyleColor(ImGuiCol_Button, button_color);
+        ImGui_ScopePushStyleColor(ImGuiCol_ButtonHovered, sensor_header_light_blue); //TODO: Change color?
+        ImGui_ScopePushStyleColor(ImGuiCol_ButtonActive, regular_blue); //TODO: Change color?
+        ImGui_ScopePushStyleColor(ImGuiCol_Text, light_grey);
+        ImGui_ScopePushStyleColor(ImGuiCol_TextSelectedBg, light_grey);
+        ImGui_ScopePushStyleColor(ImGuiCol_TitleBg, header_color);
+        ImGui_ScopePushStyleColor(ImGuiCol_PopupBg, sensor_bg);
+        ImGui_ScopePushStyleVar(ImGuiStyleVar_WindowPadding, ImVec2(5, 5));
+        auto close_clicked = false;
+        //ImGui_ScopePushStyleVar(ImGuiAlign, ImGuiAlign_Center);
+        ImGui::OpenPopup(title.c_str());
+        if (ImGui::BeginPopupModal(title.c_str(), NULL, ImGuiWindowFlags_AlwaysAutoResize))
+        {
+            ImGui::NewLine();
+            ImGui::Text("%s", process_topic_text.c_str());
+            ImGui::NewLine();
+            auto process_topic_text_size = ImGui::CalcTextSize(process_topic_text.c_str()).x ;
+            auto process_status_text_size = ImGui::CalcTextSize(process_status_text.c_str()).x + ImGui::CalcTextSize("Status: ").x;
+            if (process_topic_text_size > process_status_text_size)
+                ImGui::SetCursorPosX(ImGui::GetCursorPosX() + process_topic_text_size / 2.f - process_status_text_size / 2.f);
+
+            ImGui::Text("Status: %s", process_status_text.c_str());
+            ImGui::NewLine();
+            auto width = ImGui::GetWindowWidth();
+
+            if (enable_close)
+            {
+
+                ImGui::SetCursorPosX(ImGui::GetCursorPosX() + width / 2.f - 30.f);
+                if (ImGui::Button("Close", ImVec2(60, 30)))
+                {
+                    ImGui::CloseCurrentPopup();
+                    close_clicked = true;
+                }
+            }
+
+            ImGui::EndPopup();
+        }
+        return close_clicked;
+    }
+
+    bool device_model::prompt_toggle_advanced_mode(bool enable_advanced_mode, const std::string& message_text, std::vector<std::string>& restarting_device_info, viewer_model& view, ux_window& window, std::string& error_message)
```

Comment:
Why is the error message in these new calls passed as non-const?
Same for restarting_device_info -- changed?

Reply:
It is an addition, but it could be const. I will add a const specifier/